rebasehelper/build_tools/koji_tool.py

```diff
@@ -102,10 +109,28 @@ def _scratch_build(cls, source, **kwargs):
             logger.info('RPM build failed %s', weburl)
             logs.append(weburl)
             cls.logs.append(weburl)
-            raise BinaryPackageBuildError
+            return_code = cls.get_mock_return_code(build_error)
+            if return_code:
+                raise BinaryPackageBuildError(return_code=return_code)
+            else:
+                # Unknown error message obtained from koji
+                raise BinaryPackageBuildError()
         logs.append(weburl)
         return rpms, logs, task_id
 
+    @classmethod
+    def get_mock_return_code(cls, msg):
+        """
+        Parses mock status message for mock build return code
+
+        :param msg: mock status message
+        :return: returns mock build return code
+        """
+        if re.search(r'status [0-9]+', msg):
+            found = re.search(r'[0-9]+', msg)
```

Comment:
This could definitely be done with just one `re.search` call. I would also rather match for more specific string, like `mock exited with status \d+`.
Also it would be nice to include a comment with typical error message for which the regexp is valid, to be able to update it in case the message changes in the future.